corehq/celery_monitoring/signals.py

```diff
@@ -22,6 +24,30 @@ def celery_add_time_sent(headers=None, body=None, **kwargs):
     TimeToStartTimer(task_id).start_timing(eta)
 
 
+@after_task_publish.connect
+def update_celery_state(sender=None, headers=None, **kwargs):
+    """Updates the celery task progress to "SENT"
+
+    When fetching an task from celery using the form AsyncResponse(task_id), if
+    task_id never exists, celery will not throw an error, it will just hang.
+    This function updates each task sent to celery to have a progress value of "SENT",
+    which means we can check that a task exists with the following pattern:
+    `AsyncResponse(task_id).status == "SENT"`
+
+    See
+    http://stackoverflow.com/questions/9824172/find-out-whether-celery-task-exists/10089358
+    for more info
+
+    """
+
+    # ensure connection to DB is usable
+    close_old_connections()
```

Comment:
I was seeing "connection already closed" errors in test class cleanup callbacks when this was being called on test teardown before the cleanup callbacks were executed. I don't quite understand why that was happening, but my guess is because the cleanup callbacks were somehow referencing the "old" connections that were closed.

Wondering if this be the last thing done this function rather than the first?

Reply:
59f588e64e16c702b2afccc526808a5d01ebe380